### PR TITLE
Support pgsql-kafka binding

### PR DESCRIPTION
**Pull request by akrambek**
## Description

Support pgsql-kafka binding.

Fixes #[1058](https://github.com/aklivity/zilla/issues/1058)

### CODE REVIEW

**Review by jfallows — DISMISSED** [review]

File: incubator/binding-pgsql-kafka/pom.xml [hunk 1]
```diff
@@ -191,6 +195,7 @@
         <configuration>
           <excludes>
             <exclude>io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/types/**/*.class</exclude>
+            <exclude>net/sf/jsqlparser/parser/*</exclude>
```
Comment (jfallows) [inline incubator/binding-pgsql-kafka/pom.xml:198]: Why is this needed? Does `jsqlparser` generate some code?

File: incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/config/PgsqlKafkaBindingConfig.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        PgsqlKafkaConfiguration config,
+        BindingConfig binding,
+        LongFunction<CatalogHandler> supplyCatalog)
+    {
+        this.id = binding.id;
+        this.name = binding.name;
+        this.kind = binding.kind;
+        this.routes = binding.routes.stream().map(PgsqlKafkaRouteConfig::new).collect(toList());
+
+        this.catalog = supplyCatalog.apply(binding.catalogs.get(0).id);
```
Comment (jfallows) [inline incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/config/PgsqlKafkaBindingConfig.java:47]: What is `catalog` is omitted?
Perhaps we can make it `required` in the schema patch?

File: incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/PgsqlKafkaConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    public static final ShortPropertyDef KAFKA_CREATE_TOPICS_REPLICAS;
+    public static final PropertyDef KAFKA_AVRO_SCHEMA_NAMESPACE;
+
+    static
+    {
+        final ConfigurationDef config = new ConfigurationDef(String.format("zilla.binding.%s", PgsqlKafkaBinding.NAME));
+        KAFKA_TOPIC_REQUEST_TIMEOUT_MS = config.property("kafka.topic.request.timeout.ms", 30000);
+        KAFKA_CREATE_TOPICS_PARTITION_COUNT = config.property("kafka.create.topics.partition.count", 1);
+        KAFKA_CREATE_TOPICS_REPLICAS = config.property("kafka.create.topics.replicas", (short) 1);
+        KAFKA_AVRO_SCHEMA_NAMESPACE = config.property("kafka.avro.schema.namespace", "io.aklivity.zilla");
```
Comment (jfallows) [inline incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/PgsqlKafkaConfiguration.java:34]: This doesn't seem like the right value for the avro namespace as it would more naturally be part of the application, not part of zilla directly.

Seems like it should perhaps default to the pgsql `database` parameter instead, like `dev` and perhaps this can be a pattern that would substitute the pgsql `database` parameter into it, so we could configure it with something like `"com.example.{database}"` and have it resolve to `"com.example.dev"`. Then if there is no reference to `{database}` in the pattern, it would be used verbatim as it is now.

File: incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/stream/PgsqlKafkaCommandType.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        for (PgsqlKafkaCommandType commandType : PgsqlKafkaCommandType.values())
+        {
+            if (Arrays.equals(commandType.value, value))
+            {
+                command = commandType;
+                break command;
+            }
+        }
+
+        return command;
```
Comment (jfallows) [inline incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/stream/PgsqlKafkaCommandType.java:51]: Each call to `PgsqlKafkaCommandType.values()` reallocates the array of enums.

Perhaps we can use a `static { ... }` initializer to create a static map of `byte[]` to `PgsqlKafkaCommandType` and then use that map here to lookup the `byte[]`?

File: incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/stream/PgsqlKafkaProxyFactory.java [hunk 5]
```diff
@@ -0,0 +1,1444 @@
...
+
+        MessageConsumer receiver =
+            streamFactory.newStream(begin.typeId(), begin.buffer(), begin.offset(), begin.sizeof(), sender);
+
+        receiver.accept(begin.typeId(), begin.buffer(), begin.offset(), begin.sizeof());
+
+        return receiver;
+    }
+
+    private void onDecodeCreateTopicCommand(
```
Comment (jfallows) [inline incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/stream/PgsqlKafkaProxyFactory.java:1322]:
```suggestion
    private void decodeCreateTopicCommand(
```

File: incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/stream/PgsqlKafkaProxyFactory.java [hunk 6]
```diff
@@ -0,0 +1,1444 @@
...
+            final String policy = binding.avroValueSchema.primaryKey(statement) != null
+                ? "compact"
+                : "delete";
+
+            final KafkaCreateTopicsProxy createTopicsProxy = server.createTopicsProxy;
+            createTopicsProxy.doKafkaBegin(traceId, authorization, topics, policy);
+        }
+    }
+
+    private void onDecodeUnknownCommand(
```
Comment (jfallows) [inline incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/stream/PgsqlKafkaProxyFactory.java:1360]:
```suggestion
    private void decodeUnknownCommand(
```

**Review by jfallows — CHANGES_REQUESTED** [review]

File: incubator/binding-pgsql-kafka/pom.xml (hunk 1 above)
Comment (akrambek) [reply incubator/binding-pgsql-kafka/pom.xml:198]: No but it was causing some IlligalClassFormatException in jacoco

```
java.lang.instrument.IllegalClassFormatException: Error while instrumenting net/sf/jsqlparser/parser/CCJSqlParserTokenManager with JaCoCo 0.8.11.202310140853/f33756c.
	at org.jacoco.agent.rt.internal_4742761.CoverageTransformer.transform(CoverageTransformer.java:94)
	at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:244)
	at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:541)
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1013)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:862)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:760)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:681)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:639)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at net.sf.jsqlparser.parser.CCJSqlParser.<init>(CCJSqlParser.java:42028)
	at net.sf.jsqlparser.parser.CCJSqlParserManager.parse(CCJSqlParserManager.java:21)
```

File: incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/PgsqlKafkaConfiguration.java (hunk 3 above)
Comment (akrambek) [reply incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/PgsqlKafkaConfiguration.java:34]: Yeah, good suggestion. I was not sure what you recommend since com.example was too generic

**Review by jfallows — APPROVED** [review]

**Review by jfallows — APPROVED** [review]
